### PR TITLE
Add enwiki-20200101 streaming dataset

### PR DESCRIPTION
**Pull request by knighton**
Command to perform conversion to MDS:

streaming$ python3 -m streaming.text.convert.enwiki --in_root /workdisk/vlad/mlperf-bert-nvidia/workspace/bert_data/download/results4/ --out_root /workdisk/james/enwiki/

### CODE REVIEW

**Comment by dskhudia** [comment]
Can you add the command you used to call this script in the description so that it's available for future?

File: streaming/text/convert/enwiki.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        default=0,
+        help='Keeps all traces of the progressbar upon termination of iteration. Default: 0 ' +
+        '(False)',
+    )
+    return args.parse_args()
+
+
+def process_split(in_root: str, out_root: str, compression: str, hashes: List[str], size_limit: int,
+                  progbar: int, leave: int, basenames: List[str], split: str) -> None:
+    """Process a dataset split.
```
Comment (dskhudia) [inline streaming/text/convert/enwiki.py:66]: What does this mean? Process a dataset that has already been split to different shards?

**Review by dskhudia — APPROVED** [review]

**Comment by karan6181** [comment]
Looks good overall. Can you please address below two points? Thanks!
1. Add a description to the PR on what the PR is about
2. `pre-commit` is failing with unused import and Missing docstring.

File: streaming/text/convert/enwiki.py (hunk 1 above)
Comment (karan6181) [reply streaming/text/convert/enwiki.py:66]: No. The `process_split()` reads the raw dataset (for example, which we downloaded from internet) and converts into `*.mds` file extension with multiple shard files.

Comment (karan6181) [reply streaming/text/convert/enwiki.py:66]: @knighton maybe it needs a better descriptive method comments ?

Comment (dskhudia) [reply streaming/text/convert/enwiki.py:66]: In this case basenames is already a collection of 500 txt files.

Comment (dskhudia) [reply streaming/text/convert/enwiki.py:66]: Got it. So it's just reading the files one-by-one but sharding etc. is handled by the writer.

File: streaming/text/convert/enwiki.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        '--hashes',
+        type=str,
+        default='sha1,xxh64',
+        help='Hashing algorithms to apply to shard files. Default: sha1,xxh64',
+    )
+    args.add_argument(
+        '--size_limit',
+        type=int,
+        default=1 << 25,
+        help='Shard size limit, after which point to start a new shard. Default: 1 << 27',
```
Comment (karan6181) [inline streaming/text/convert/enwiki.py:51]: nit: `1 << 25` in `help` message

File: streaming/text/convert/enwiki.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+def main(args: Namespace) -> None:
+    """Main: create streaming enwiki dataset.
+
+    Args:
+        args (Namespace): command-line arguments.
+    """
+    hashes = get_list_arg(args.hashes)
+
+    basenames = [f'part-{i:05}-of-00500' for i in range(500)]
```
Comment (karan6181) [inline streaming/text/convert/enwiki.py:108]: Is the input filename always has this predefined name? can there be a case where the filename is different for any user? Wondering if we need something like `os.walk()` to get the filename during runtime?

File: streaming/text/enwiki.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            raise ValueError(f'Only group_method="truncate" is supported at this time.')
+
+        super().__init__(local, remote, split, shuffle, prefetch, keep_zip, retry, timeout, hash,
+                         batch_size)
+        self.tokenizer_name = tokenizer_name
+        self.max_seq_len = max_seq_len
+        self.group_method = group_method
+
+        # Build tokenizer
+        self.tokenizer = AutoTokenizer.from_pretrained(self.tokenizer_name)  # pyright: ignore
```
Comment (karan6181) [inline streaming/text/enwiki.py:65]: is this `# pyright: ignore` still relevant here of copy/paste ?

File: streaming/text/convert/enwiki.py (hunk 2 above)
Comment (knighton) [reply streaming/text/convert/enwiki.py:51]: Done

File: streaming/text/convert/enwiki.py (hunk 3 above)
Comment (knighton) [reply streaming/text/convert/enwiki.py:108]: Yes, always named as such, as the output of a previous preprocessing stage

File: streaming/text/enwiki.py (hunk 4 above)
Comment (knighton) [reply streaming/text/enwiki.py]: Will try removing it

Comment (karan6181) [reply streaming/text/enwiki.py]: Fixed in https://github.com/mosaicml/streaming/pull/23/commits/926d9779baed22f1e9392a6b96b43cfc3f63d68e

File: streaming/text/convert/enwiki.py (hunk 2 above)
Comment (karan6181) [reply streaming/text/convert/enwiki.py]: Fixed in https://github.com/mosaicml/streaming/pull/23/commits/926d9779baed22f1e9392a6b96b43cfc3f63d68e.

**Review by karan6181 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**